include/swift/AST/DiagnosticsSema.def

```diff
@@ -3264,6 +3264,18 @@ ERROR(autodiff_attr_result_not_differentiable,none,
       "'Differentiable', but %0 does not conform to 'Differentiable'", (Type))
 ERROR(autodiff_attr_opaque_result_type_unsupported,none,
       "cannot differentiate functions returning opaque result types", ())
+ERROR(autodiff_attr_static_decl_mismatch,none,
+      "derivative function %0 operates on %select{an instance|a}1 type, "
+      "not on %select{an instance|a}2 type as required",
+      (/*derivative*/ DeclName, /*derivativeOperatesOnTy*/bool,
+       /*originalOperatesOnTy*/bool))
+NOTE(autodiff_attr_static_decl_original,none,
+      "original function %0 operates on %select{an instance|a}1 type",
```

Comment:
I'm not sure "operates on a type" is the right user-facing error message, as it sounds quite uncommon in existing Swift diagnostics. Plus, "an instance type" doesn't sound quite right — shouldn't it just be "an instance"?

Instead of "operates on something", how about something like "original function 'foo(_:)' is {an instance|a static|a class} method"?

Reply:
Indeed, your suggestion seems more aligned with wording already existing in the diagnostic error messages. 
I'll make this note disappear and include your suggestion in the `error` diagnostic itself. Thanks!